src/printer.js

```diff
@@ -2285,6 +2285,16 @@ function printMemberChain(path, options, print) {
     if (
       hasSeenCallExpression && printedNodes[i].node.type === "MemberExpression"
     ) {
+      // [0] should be appended at the end of the group instead of the
+      // beginning of the next one
+      if (printedNodes[i].node.computed &&
+          printedNodes[i].node.property.type === "NumericLiteral" ||
+          printedNodes[i].node.property.type === "Identifier" &&
+          typeof printedNodes[i].node.property.value === "number") {
```

Comment:
I feel like this is too specific. It seems like any computed property should be part of the previous group. I can't think of any scenario where I'd want the brackets to be on the next line.